pkg/postgres/pgutils/error.go

```diff
@@ -46,6 +52,20 @@ func isTransientError(err error) bool {
 	if pgErr := (*pgconn.PgError)(nil); errors.As(err, &pgErr) {
 		return transientPGCodes.Contains(pgErr.Code)
 	}
-	// Assume all other errors are transient
-	return false
+	if pgconn.SafeToRetry(err) {
+		return true
+	}
+	switch err {
+	case pgx.ErrTxClosed, pgx.ErrTxCommitRollback:
```

Comment:
This doesn't feel retryable. It could be any reason within the txn that caused it to fail.